test/runtests.jl

```diff
@@ -16,67 +16,69 @@ end
 
 for (M, f, arity) in DiffRules.diffrules(; filter_modules=nothing)
-    (M, f, arity) ∈ non_diffeable_arg_functions && continue
-    if arity == 1
-        @test DiffRules.hasdiffrule(M, f, 1)
-        deriv = DiffRules.diffrule(M, f, :goo)
-        modifier = if f in (:asec, :acsc, :asecd, :acscd, :acosh, :acoth)
-            1.0
-        elseif f === :log1mexp
-            -1.0
-        elseif f === :log2mexp
-            -0.5
-        else
-            0.0
-        end
-        @eval begin
-            let
-                goo = rand() + $modifier
-                @test isapprox($deriv, finitediff($M.$f, goo), rtol=0.05)
-                # test for 2pi functions
-                if "mod2pi" == string($M.$f)
-                    goo = 4pi + $modifier
-                    @test NaN === $deriv
-                end
+    for T in [Float32, Float64]
+        (M, f, arity) ∈ non_diffeable_arg_functions && continue
+        if arity == 1
+            @test DiffRules.hasdiffrule(M, f, 1)
+            deriv = DiffRules.diffrule(M, f, :goo)
+            modifier = if f in (:asec, :acsc, :asecd, :acscd, :acosh, :acoth)
+                1.0
+            elseif f === :log1mexp
+                -1.0
+            elseif f === :log2mexp
+                -0.5
+            else
+                0.0
             end
-        end
-    elseif arity == 2
-        @test DiffRules.hasdiffrule(M, f, 2)
-        derivs = DiffRules.diffrule(M, f, :foo, :bar)
-        @eval begin
-            let
-                if "mod" == string($M.$f)
-                    foo, bar = rand() + 13, rand() + 5 # make sure x/y is not integer
-                else
-                    foo, bar = rand(1:10), rand()
+            @eval begin
+                let
+                    goo = $T(rand() + $modifier)
+                    @test isapprox($deriv, finitediff($M.$f, goo), rtol=0.05)
+                    # test for 2pi functions
+                    if "mod2pi" == string($M.$f)
+                        goo = 4pi + $modifier
+                        @test NaN === $deriv
+                    end
                 end
-                dx, dy = $(derivs[1]), $(derivs[2])
-                if !(isnan(dx))
-                    @test isapprox(dx, finitediff(z -> $M.$f(z, bar), float(foo)), rtol=0.05)
-                end
-                if !(isnan(dy))
-                    @test isapprox(dy, finitediff(z -> $M.$f(foo, z), bar), rtol=0.05)
+            end
+        elseif arity == 2
+            @test DiffRules.hasdiffrule(M, f, 2)
+            derivs = DiffRules.diffrule(M, f, :foo, :bar)
+            @eval begin
+                let
+                    if "mod" == string($M.$f)
+                        foo, bar = $T(rand() + 13), $T(rand() + 5) # make sure x/y is not integer
+                    else
+                        foo, bar = rand(1:10), $T(rand())
+                    end
+                    dx, dy = $(derivs[1]), $(derivs[2])
```

Comment:
Same here check the type of `dx` and `dy`?

Reply:
It's a bit more difficult to easily address because `foo` is sometimes an integer, e.g. what do we do for `*`?

Reply:
We could only perform type checks if it's not an integer?

Reply:
Yeah, already working on this locally:)

Reply:
Added.